hpc0.qmd

```diff
@@ -1,297 +1,1300 @@
 ```{.bash code-line-numbers=false}
-[user@host ~]$ for (( i=1 ; i<=5 ; i++ ))
-> do
-> mkdir calculation_$i
-> cd calculation_$i
-> done
-[user@host calculation_5]$
+chmod 700 test0/test/test-file
 ```
+- Don't worry about remembering details of how to change permissions.
+- Try to remember that there are 3 levels of permission:
+  - user (owner), group and all
+- Try to remember that there are 3 types of permission:
+  - read, write and execute
+
+::: {.notes}
+- The following would add user (owner) execute permissions.
+  - So if the file already had all other permissions as wanted this would work. 
+```{.bash code-line-numbers=false}
+chmod u+x test0/test/test-file
+```
+- octal value = read (4) + write (2) + execute (1)
+- Later on we will set permissions on a file to execute it.
+:::
 
-In the second case, two commands are executed each time you go through the loop. A new directory is created and
-then the working directory is changed to the new directory.
 
-On the next iteration, the new directory is then created inside the previous one.
+## Groups {.smaller} {visibility="hidden"}
+- System users belong to groups (at least one).
+- A user can change the group assignment for a file or directory to other groups they belong to using the command `chgrp`.
+- The command `groups` can be used to print out what groups there are, and what group a user is part of.
+- This can be useful if you want to share files between those in a group on the system.
 
+
+## Check point {.smaller}
+- Any questons?
+
+::: {.notes}
+- Time for a break?
 :::
 
-## Exercise 6 {.smaller}
+
+## Variables {.smaller}
+- Variables are defined and accessed as follows:
+```{.bash code-line-numbers=false}
+var=1
+echo $var
+```
+- The variable `var` is set equal to the number `1`.
+- `echo` is a command that will send what comes after it to standard out
+- The value of the variable is accessed using the `$` symbol followed by the name of the variable.
+- The command `printenv` will print all the variables.
+
+
+## For loop 1/2 {.smaller}
+```
+- Example for loop with iterator syntax to print the numbers 1 to 5:
+```{.bash code-line-numbers=false}
+for ((i=1;i<=5;i=i+1)) do echo $i; done
+```
+- Example for loop with sequence syntax to print the numbers 1 to 5:
+```{.bash code-line-numbers=false}
+for i in {1..5}; do echo $i; done
+```
+- Example for loop with sequence syntax to print even numbers from 2 to 10:
+```{.bash code-line-numbers=false}
+for i in {2..10..2}; do echo $i; done
+```
+- Example for loop with iterator syntax to print a geometric series from 2 to 256:
+```{.bash code-line-numbers=false}
+for ((i=2;i<=265;i=i*2)) do echo $i; done
+```
+
+
+## For loop 2/2 {.smaller}
+- Example for loop to loop through directory contents:
+```{.bash code-line-numbers=false}
+for f in *; do echo $f; done
+```
+- Where there is a '`;`' there can be a new line:
+```{.bash code-line-numbers=false}
+for f in *
+ do echo $f
+ done
```

Comment:
```suggestion
done
```

Reply:
Actually, do we indent things at this stage, or wait until you introduce shell scripts?  Whatever, line up with the solutions later on.